cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/internal/CouchDbUtil.java

```diff
@@ -127,12 +127,15 @@ public static String removeExtension(String fileName) {
     }
 
     public static String streamToString(InputStream in) {
-        Scanner s = new Scanner(in, "UTF-8");
-        s.useDelimiter("\\A");
-        String str = s.hasNext() ? s.next() : null;
-        close(in);
-        s.close();// mdb
-        return str;
+        try {
+            Scanner s = new Scanner(in, "UTF-8");
+            s.useDelimiter("\\A");
+            String str = s.hasNext() ? s.next() : null;
+            s.close();// mdb
+            return str;
+        } finally {
+            close(in);
```

Comment:
This `close()` method here is a static in `CouchDbUtil` that does call `IOUtils.closeQuietly()`. I agree in some places it would be better to do the raw close (e.g. in `HttpConnection` where we are already can throw a checked `IOException`). That said until we move up the Java level to be able to use try-with-resources I'm reluctant to add all the levels of nesting required to perform the necessary `null` checks and multiple try/finally blocks especially where we are trying to close multiple `Closeable`s.